iron-list.html

```diff
@@ -209,6 +216,34 @@
       },
 
       /**
+       * When true, `iron-list` will render the items in a fixed-width grid.
+       * If true the `itemWidth` property will also need to be set
+       */
+      grid: {
+        type: Boolean,
+        value: false,
+        reflectToAttribute: true
+      },
+
+      /**
+       * When `grid` is true this will define the width of each item in the
+       * grid. This plus the value of `itemOffset` will determine how many
+       * items are rendered per row
+       */
+      itemWidth: {

```

Comment:
I'm totally down for this. My only problem is that it would need to be clear (as you indicated in your sample code) that the width is being determined off of one of the items (I agree it should be the first). I actually thought about making the `itemWidth` prop optional and defaulting to the first item width if it's not passed.

To take this a step further we could also grab the margin left/right and use that for the offset instead if the `itemOffset` is not passed.